src/githubClient.ts

```diff
@@ -6,7 +8,35 @@ export class GitHubClient {
         this._token = token;
     }
 
-    public async createCheckRun(payload: any): Promise<void> {
+    public async createCheckRunWithFallback(checkRunThroughAppPayload: any, checkRunPayload: any): Promise<void> {
+        const checkRunUrl = `https://api.github.com/repos/${this._repository}/container-scanning/check-run`;
+        const webRequest = new WebRequest();
+        webRequest.method = "POST";
+        webRequest.uri = checkRunUrl;
+        webRequest.body = JSON.stringify(checkRunThroughAppPayload);
+        webRequest.headers = {
+            Authorization: `Bearer ${this._token}`
+        };
+
+        console.log(`Creating check run. image_name: ${checkRunThroughAppPayload['image_name']}, head_sha: ${checkRunThroughAppPayload['head_sha']}`);
+        const response: WebResponse = await sendRequest(webRequest);
+        core.debug(util.format('Response from scanitizer app:\n', response.body));
+        if (response.statusCode == StatusCodes.UNPROCESSABLE_ENTITY
+            && response.body
+            && response.body.message
+            && response.body.message.error_code === 'APP_NOT_INSTALLED') {
+            console.log('Looks like the scanitizer app is not installed on the repo. Falling back to check run creation through GitHub actions app...');
```

Comment:
Get this reviewed by PM. Also, we need to figure out a different entry point for this. This won't be discoverable in a plethora of logs!

Reply:
cc: @raiyanalam

Reply:
cc: @azooinmyluggage

Reply:
Filed an issue to track this:
https://github.com/Azure/container-scan/issues/22